### ISSUE TITLE
Add `RCropSize` for a randomly placed `CropSize`

### ISSUE THREAD

**Issue by Evizero** [Evizero/Augmentor.jl#21]
for a bit more variation

**Comment by johnnychen94** [comment]
As a note to myself, here's something I've just made in my project

```julia
function generate_data(img;
                       noise_level=0.1,
                       patch_size=41)

    pl = ConvertEltype(Gray{Float32}) |>
         Either(NoOp(), Rotate90(), Rotate180(), Rotate270()) |>
         Either(0.5=>NoOp(), 0.25=>FlipX(), 0.25=>FlipY())
    noise = AdditiveWhiteGaussianNoise(noise_level)
    
    # randomly crop a valid 41*41 area
    w, h = map(collect.(axes(img))) do idx
        rand_idx = randperm(maximum(idx)-minimum(idx))
        idx_start = findfirst(x->x+patch_size<=maximum(idx), idx)
        idx_start = idx[idx_start]
        return idx_start:idx_start+patch_size-1
    end
    patch = augment(img, pl)[w, h]
    apply_noise!(patch, noise), patch # input, label
end
```

There're two places not so good/concise:

* move `noise` into pl via `MapFun` #35 
* a random crop function

With regard to crop, probably we're making too many crop types here 😂 I guess some can be combined into one. Will take a look when I find some time.

**Comment by johnnychen94** [comment]
closed by #64
